Purchases/Attribution/RCAttributionFetcher.m

```diff
@@ -137,15 +137,32 @@ - (BOOL)isAuthorizedToPostSearchAds {
 
         BOOL needsTrackingAuthorization = [self.systemInfo isOperatingSystemAtLeastVersion:minimumOSVersionRequiringAuthorization];
 
-        Class<FakeATTrackingManager> _Nullable trackingManagerClass = [self.attributionFactory atTrackingManagerClass];
-        if (!trackingManagerClass && needsTrackingAuthorization) {
-            RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+        Class _Nullable trackingManagerClass = [self.attributionFactory atTrackingClass];
+        if (!trackingManagerClass) {
+            if (needsTrackingAuthorization) {
+                RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+            }
+            return !needsTrackingAuthorization;
```

Comment:
this was sort of missing in the previous implementation: if the tracking class is not there and you don't need authorization, the code would still run to the end, although I don't think it was a problem since it would have been interpreted as value = 0, auth status not determined